src/main/scala/com/etsy/sbt/Checkstyle.scala

```diff
@@ -10,56 +10,110 @@ import sbt._
 
 /**
   * An SBT plugin to run checkstyle over Java code
-  *  
+  *
   * @author Andrew Johnson <ajohnson@etsy.com>
+  * @author Alejandro Rivera <alejandro.rivera.lopez@gmail.com>
+  * @author Joseph Earl <joe@josephearl.co.uk>
   */
 object Checkstyle extends Plugin {
+
   import com.etsy.sbt.Checkstyle.CheckstyleTasks._
 
   object CheckstyleTasks {
     val checkstyle = TaskKey[Unit]("checkstyle", "Runs checkstyle")
+    val checkstyleCheck = TaskKey[Unit]("checkstyle-check", "Runs checkstyle and fails the task if issues are found")
     val checkstyleTarget = SettingKey[File]("checkstyle-target", "The location of the generated checkstyle report")
     val checkstyleConfig = SettingKey[File]("checkstyle-config", "The location of the checkstyle configuration file")
     val xsltTransformations = SettingKey[Option[Set[XSLTSettings]]]("xslt-transformations", "An optional set of XSLT transformations to be applied to the checkstyle output")
+    val checkstyleCheckSeverityLevel = SettingKey[String]("checkstyle-check-level", "Sets the severity levels which should fail the build")
```

Comment:
What are your thoughts on defining an enum/case objects to enumerate the possible set of valid values here instead of using a string?